okhttp/src/main/java/okhttp3/Challenge.java

```diff
@@ -17,14 +17,16 @@
 
 import okhttp3.internal.Util;
 
-/** An RFC 2617 challenge. */
+/** An RFC 7617 challenge. */
 public final class Challenge {
   private final String scheme;
   private final String realm;
+  private final String charset;
 
-  public Challenge(String scheme, String realm) {
+  public Challenge(String scheme, String realm, String charset) {
```

Comment:
We definitely need to keep the original constructor for compatibility. I was thinking something like this:

```
Challenge c = new Challenge("basic", "WallyWorld");
c = c.withCharset(StandardCharsets.UTF_8);
```

Basically keep the public constructor as-is and add a public factory method that adds a charset to an existing challenge.